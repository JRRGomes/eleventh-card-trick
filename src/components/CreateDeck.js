const SUITS = ["H", "S", "D", "C"];
const CARDS = [
  "A",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "J",
  "Q",
  "K",
];

const CreateDeck = () => {
  let deck = SUITS.flatMap((suit) => 
  CARDS.map(card => ({suit: suit, card: card})
  ))
  return deck
}
export default CreateDeck
