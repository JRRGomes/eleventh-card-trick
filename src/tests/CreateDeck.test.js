import CreateDeck from "../components/CreateDeck";

describe("CreateDeck", () => {
  it("should return a array of objects with 52 cards/suits", () => {
    expect(CreateDeck()).toMatchSnapshot();
  });
});
