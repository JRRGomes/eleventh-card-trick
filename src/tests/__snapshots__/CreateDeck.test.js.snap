// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDeck should return a array of objects with 52 cards/suits 1`] = `
Array [
  Object {
    "card": "A",
    "suit": "H",
  },
  Object {
    "card": "2",
    "suit": "H",
  },
  Object {
    "card": "3",
    "suit": "H",
  },
  Object {
    "card": "4",
    "suit": "H",
  },
  Object {
    "card": "5",
    "suit": "H",
  },
  Object {
    "card": "6",
    "suit": "H",
  },
  Object {
    "card": "7",
    "suit": "H",
  },
  Object {
    "card": "8",
    "suit": "H",
  },
  Object {
    "card": "9",
    "suit": "H",
  },
  Object {
    "card": "10",
    "suit": "H",
  },
  Object {
    "card": "J",
    "suit": "H",
  },
  Object {
    "card": "Q",
    "suit": "H",
  },
  Object {
    "card": "K",
    "suit": "H",
  },
  Object {
    "card": "A",
    "suit": "S",
  },
  Object {
    "card": "2",
    "suit": "S",
  },
  Object {
    "card": "3",
    "suit": "S",
  },
  Object {
    "card": "4",
    "suit": "S",
  },
  Object {
    "card": "5",
    "suit": "S",
  },
  Object {
    "card": "6",
    "suit": "S",
  },
  Object {
    "card": "7",
    "suit": "S",
  },
  Object {
    "card": "8",
    "suit": "S",
  },
  Object {
    "card": "9",
    "suit": "S",
  },
  Object {
    "card": "10",
    "suit": "S",
  },
  Object {
    "card": "J",
    "suit": "S",
  },
  Object {
    "card": "Q",
    "suit": "S",
  },
  Object {
    "card": "K",
    "suit": "S",
  },
  Object {
    "card": "A",
    "suit": "D",
  },
  Object {
    "card": "2",
    "suit": "D",
  },
  Object {
    "card": "3",
    "suit": "D",
  },
  Object {
    "card": "4",
    "suit": "D",
  },
  Object {
    "card": "5",
    "suit": "D",
  },
  Object {
    "card": "6",
    "suit": "D",
  },
  Object {
    "card": "7",
    "suit": "D",
  },
  Object {
    "card": "8",
    "suit": "D",
  },
  Object {
    "card": "9",
    "suit": "D",
  },
  Object {
    "card": "10",
    "suit": "D",
  },
  Object {
    "card": "J",
    "suit": "D",
  },
  Object {
    "card": "Q",
    "suit": "D",
  },
  Object {
    "card": "K",
    "suit": "D",
  },
  Object {
    "card": "A",
    "suit": "C",
  },
  Object {
    "card": "2",
    "suit": "C",
  },
  Object {
    "card": "3",
    "suit": "C",
  },
  Object {
    "card": "4",
    "suit": "C",
  },
  Object {
    "card": "5",
    "suit": "C",
  },
  Object {
    "card": "6",
    "suit": "C",
  },
  Object {
    "card": "7",
    "suit": "C",
  },
  Object {
    "card": "8",
    "suit": "C",
  },
  Object {
    "card": "9",
    "suit": "C",
  },
  Object {
    "card": "10",
    "suit": "C",
  },
  Object {
    "card": "J",
    "suit": "C",
  },
  Object {
    "card": "Q",
    "suit": "C",
  },
  Object {
    "card": "K",
    "suit": "C",
  },
]
`;
